commit 5f8dc33e8ee7e59bee3bc6dc2088807a384b285a
Author: Nikolay Aleksandrov <nikolay@cumulusnetworks.com>
Date:   Fri Nov 13 14:54:01 2015 +0100

    net: fix feature changes on devices without ndo_set_features
    
    When __netdev_update_features() was updated to ensure some features are
    disabled on new lower devices, an error was introduced for devices which
    don't have the ndo_set_features() method set. Before we'll just set the
    new features, but now we return an error and don't set them. Fix this by
    returning the old behaviour and setting err to 0 when ndo_set_features
    is not present.
    
    Fixes: e7868a85e1b2 ("net/core: ensure features get disabled on new lower devs")
    CC: Jarod Wilson <jarod@redhat.com>
    CC: Jiri Pirko <jiri@resnulli.us>
    CC: Ido Schimmel <idosch@mellanox.com>
    CC: Sander Eikelenboom <linux@eikelenboom.it>
    CC: Andy Gospodarek <gospo@cumulusnetworks.com>
    CC: Florian Fainelli <f.fainelli@gmail.com>
    Signed-off-by: Nikolay Aleksandrov <nikolay@cumulusnetworks.com>
    Reviewed-by: Jiri Pirko <jiri@mellanox.com>
    Reviewed-by: Andy Gospodarek <gospo@cumulusnetworks.com>
    Reviewed-by: Jarod Wilson <jarod@redhat.com>
    Tested-by: Florian Fainelli <f.fainelli@gmail.com>
    Tested-by: Dave Young <dyoung@redhat.com>
    Tested-by: Christian Borntraeger <borntraeger@de.ibm.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

--- a/net/core/dev.c
+++ b/net/core/dev.c
@@ -6359,6 +6359,8 @@ int __netdev_update_features(struct net_
 
 	if (dev->netdev_ops->ndo_set_features)
 		err = dev->netdev_ops->ndo_set_features(dev, features);
+	else
+		err = 0;
 
 	if (unlikely(err < 0)) {
 		netdev_err(dev,
